### PR TITLE
Ref #541 -- Allow delete waitinglist entries

### PR DESCRIPTION
**Pull request by danlls**
Add new action "delete" for each waiting list entry

### CODE REVIEW

File: src/pretix/control/templates/pretixcontrol/waitinglist/index.html [hunk 1]
```diff
@@ -96,6 +96,7 @@
                     <th>{% trans "On the list since" %}</th>
                     <th>{% trans "Status" %}</th>
                     <th>{% trans "Voucher" %}</th>
+                    <th>{% trans "Delete" %}</th>
```
Comment (raphaelm) [inline src/pretix/control/templates/pretixcontrol/waitinglist/index.html]: We normally do not label the last column, if it only contains buttons that speak for themselves, so just use ``<th></th>``

File: src/pretix/control/views/waitinglist.py [hunk 2]
```diff
@@ -72,6 +72,27 @@ def post(self, request, *args, **kwargs):
...
+                return redirect(reverse('control:event.orders.waitinglist', kwargs={
+                    'event': request.event.slug,
+                    'organizer': request.event.organizer.slug
+                }))
+            try:
+                wle = WaitingListEntry.objects.get(
+                    pk=request.POST.get('delete'), event=self.request.event,
+                )
+                wle.delete()
+                messages.success(request, _('Entry is successfully deleted.'))
```
Comment (raphaelm) [inline src/pretix/control/views/waitinglist.py]: More consistant with the rest of the system would be "The selected entry has been deleted"

File: src/pretix/control/templates/pretixcontrol/waitinglist/index.html [hunk 3]
```diff
@@ -133,6 +134,10 @@
                                 </button>
                             {% endif %}
                         </td>
+                        <td>
+                            </button>
```
Comment (raphaelm) [inline src/pretix/control/templates/pretixcontrol/waitinglist/index.html]: What is that ``</button>`` doing there? ;)

File: src/pretix/control/templates/pretixcontrol/waitinglist/delete.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+		{% csrf_token %}
+		<p>{% blocktrans %}Are you sure you want to delete the following waiting list entry <strong>{{ entry }}</strong>?{% endblocktrans %}</p>
+		<div class="form-group submit-group">
+            <a href="{% url "control:event.orders.waitinglist" organizer=request.event.organizer.slug event=request.event.slug %}" class="btn btn-default btn-cancel">
+                {% trans "Cancel" %}
+            </a>
+            <button type="submit" class="btn btn-danger btn-save">
+                {% trans "Delete" %}
+            </button>
+		</div>
```
Comment (raphaelm) [inline src/pretix/control/templates/pretixcontrol/waitinglist/delete.html]: Please adjust the indentation here a bit ;)

**Comment by danlls** [comment]
Good point, but apparently once an entry is assigned a voucher, it no longer appears on the waiting list. Hence, it can't be deleted.

File: src/pretix/control/views/waitinglist.py [hunk 5]
```diff
@@ -125,3 +128,33 @@ def get_sales_estimate(self):
...
+class EntryDelete(EventPermissionRequiredMixin, DeleteView):
+    model = WaitingListEntry
+    template_name = 'pretixcontrol/waitinglist/delete.html'
+    permission = 'can_change_orders'
+    context_object_name = 'entry'
+
+    def get_object(self, queryset=None) -> WaitingListEntry:
+        try:
+            return self.request.event.waitinglistentries.get(
+                id=self.kwargs['entry'],
```
Comment (raphaelm) [inline src/pretix/control/views/waitinglist.py]: We'll also need to check here if there is no voucher attached, hiding the button alone is not enought. The easies way would be to just add ``voucher__isnull=True`` to this database query here

**Comment by danlls** [comment]
@raphaelm Done! :D 

*off-topic*: Can you suggest an issue for me to work on?

**Comment by raphaelm** [comment]
Thank you very much!

What do you think about #543 next?

**Pull request merged**